pyro/distributions/torch_distribution.py

```diff
@@ -167,10 +167,10 @@ def has_enumerate_support(self):
         return self.base_dist.has_enumerate_support
 
     def sample(self, sample_shape=torch.Size()):
-        return self.base_dist.sample(self.sample_shape + sample_shape)
+        return self.base_dist.sample(sample_shape + self.sample_shape)
```

Comment:
what is the difference between this change and the way it was?

Reply:
The desired behavior is for `.reshape()` to add shape on the left, and `.sample()` to add shape on the left of that. Thus the following should be equivalent:
```py
assert dist.reshape(shape1).sample(shape2).shape \
    == dist.sample(shape2 + shape1).shape
```
For example if you want to replace a 3-dimensional multivariate normal with a diag normal, the following should have the same shape:
```py
dist1 = MultivariateNormal(variable([0, 0, 0]), ...)
dist2 = Normal(0, 1).reshape(sample_shape=3, extra_event_dims=3)  # diag normal
assert dist1.event_shape == dist2.event_shape
assert dist1.batch_shape == dist2.batch_shape
assert dist1.sample([100]).shape == (100, 3)
assert dist2.sample([100]).shape == (100, 3)
```
I've added a simple test to verify this behavior explicitly.

Reply:
thanks this is helpful for my understanding. i would have expected ```extra_event_dims=1```?